src/inventory_ui.cpp

```diff
@@ -721,86 +1149,79 @@ void inventory_selector::toggle_navigation_mode()
     }
 }
 
-void inventory_selector::insert_column( decltype( columns )::iterator position, std::unique_ptr<inventory_column> &column )
+void inventory_selector::append_column( inventory_column &column )
 {
-    assert( column != nullptr );
+    column.set_mode( navigation );
+    column.set_height( getmaxy( w_inv ) - 5 );
 
     if( columns.empty() ) {
-        column->on_activate();
-    }
-    column->set_mode( navigation );
-    columns.insert( position, std::move( column ) );
-}
-
-void inventory_selector::insert_selection_column( const std::string &id, const std::string &name )
-{
-    std::unique_ptr<inventory_column> new_column( new selection_column( id, name ) );
-
-    for( const auto &column : columns ) {
-        static_cast<selection_column *>( new_column.get() )->reserve_width_for( *column );
+        column.on_activate();
     }
 
-    if( column_can_fit( *new_column ) ) { // Insert only if it will be visible. Ignore otherwise.
-        new_column->set_custom_colors( true );
-        insert_column( columns.end(), new_column );
-    }
+    columns.push_back( &column );
 }
 
-inventory_pick_selector::inventory_pick_selector( player &u, const std::string &title,
-                                                  const item_location_filter &filter ) :
-            inventory_selector( u, title, filter ), null_location( new item_location() ) {}
+inventory_pick_selector::inventory_pick_selector( const player &p,
+        const inventory_selector_preset &preset ) :
+    inventory_selector( p, preset ), null_location( new item_location() ) {}
 
 item_location &inventory_pick_selector::execute()
 {
-    prepare_columns( false );
-
     while( true ) {
-        refresh_window();
+        update();
 
-        const std::string action = ctxt.handle_input();
-        const long ch = ctxt.get_raw_input().get_first_input();
```

Comment:
You should stick with `ctxt` as it's used elsewhere and therefore more intuitive when reading your code